tachyon/math/base/semigroups.h

```diff
@@ -147,12 +157,110 @@ class MultiplicativeSemigroup {
     }
     return g;
   }
+
+  constexpr static std::vector<G> GetSuccessivePowers(size_t size,
+                                                      const G& generator) {
+#if defined(TACHYON_HAS_OPENMP)
+    uint32_t log_size = size_t{base::bits::SafeLog2Ceiling(size)};
+    if (log_size <= kMinLogSizeForParallelization)
+#endif
+      return GetSuccessivePowersSerial(size, generator);
```

Comment:
I found the function `ComputePowersAndMulByConstSerial` which is identical to this function. How can we reuse the code?

Reply:
It would be good to replace the functions of UnivariateDomain with this function later on.
Plz, make issue or write a code-comment related to this situation.

Reply:
You need to place this function nearby `BatchPowers` it's very confusing and you're missing the comment here.

Reply:
@dongchangYoo I already replaced the functions in the third commit.
https://github.com/kroma-network/tachyon/pull/115/commits/d6664804171f7b04be7ea776ab18d93b4edcbfe2
Is this what you meant?

Reply:
@chokobole `BatchPowers` is the method of `AdditiveSemigroup` and `GetSuccessivePowers` is the method of `MultiplicativeSemiGroup`, so I think it is inevitable to separate those functions.

I'll add some comments though.